Sources/TuistDependencies/Carthage/Interactors/CarthageFrameworksInteractor.swift

```diff
@@ -0,0 +1,81 @@
+}
+
+public final class CarthageFrameworksInteractor: CarthageFrameworksInteracting {
+    private let fileHandler: FileHandling
+
+    public init(fileHandler: FileHandling = FileHandler.shared) {
+        self.fileHandler = fileHandler
+    }
+
+    public func copyFrameworks(carthageBuildDirectory: AbsolutePath, destinationDirectory: AbsolutePath) throws {
+        let alreadyInstalledFrameworks: Set<String> = try getAlreadyInstalledFrameworkNames(at: destinationDirectory)
+
+        try Platform.allCases.forEach { platform in
+            let carthagePlatfromBuildsDirectory = carthageBuildDirectory.appending(component: platform.carthageDirectory)
+            guard fileHandler.exists(carthagePlatfromBuildsDirectory) else { return }
+
+            var newInstalledFrameworks = Set<String>()
+
+            try fileHandler
+                .contentsOfDirectory(carthagePlatfromBuildsDirectory)
+                .filter { $0.isFolder && $0.extension == "framework" }
+                .compactMap { $0.components.last?.components(separatedBy: ".").first }
+                .forEach { frameworkName in
+                    let carthageBuildFrameworkPath = carthagePlatfromBuildsDirectory.appending(component: "\(frameworkName).framework")
+                    let destinationFrameworkPath = destinationDirectory.appending(components: frameworkName, platform.caseValue, "\(frameworkName).framework")
+                    try copyDirectory(from: carthageBuildFrameworkPath, to: destinationFrameworkPath)
+
+                    newInstalledFrameworks.insert(frameworkName)
+                }
+
+            try alreadyInstalledFrameworks
+                .subtracting(newInstalledFrameworks)
+                .forEach { frameworkName in
+                    let frameworkPath = destinationDirectory.appending(component: frameworkName)
+                    let frameworkPlatformPath = frameworkPath.appending(component: platform.caseValue)
+
+                    if fileHandler.exists(frameworkPlatformPath) {
+                        try fileHandler.delete(frameworkPlatformPath)
+
+                        if try fileHandler.contentsOfDirectory(frameworkPath).isEmpty {
+                            try fileHandler.delete(frameworkPath)
+                        }
+                    }
+                }
+        }
+    }
+
+    // MARK: - Helpers
+
+    private func getAlreadyInstalledFrameworkNames(at destinationDirectory: AbsolutePath) throws -> Set<String> {
+        guard fileHandler.exists(destinationDirectory) else { return Set<String>() }
+
+        return try fileHandler
+            .contentsOfDirectory(destinationDirectory)
+            .filter { $0.isFolder && $0.basename != "Build" }
+            .map(\.basename)
+            .reduce(into: Set<String>()) { $0.insert($1) }
+    }
+
+    private func copyDirectory(from fromPath: AbsolutePath, to toPath: AbsolutePath) throws {
```

Comment:
This method is in multiple files so I'd recommend extracting it into `FileHandler` and naming it:
```swift
func copyDirectory(from fromPath: AbsolutePath, to toPath: AbsolutePath, force: Bool) throws
```

Reply:
`CarthageFrameworksInteractor` has been deleted.